src/prpy/base/robot.py

```diff
@@ -284,15 +284,33 @@ def ViolatesVelocityLimits(self, traj):
         return False
 
     def _PlanWrapper(self, planning_method, args, kw_args):
-        # Call the planner.
-        traj = planning_method(self, *args, **kw_args)
 
-        # Strip inactive DOFs from the trajectory.
+        # Call the planner.
         config_spec = self.GetActiveConfigurationSpecification()
-        openravepy.planningutils.ConvertTrajectorySpecification(traj, config_spec)
+        result = planning_method(self, *args, **kw_args)
+
+        # Define the post processing steps for the trajectory.
+        def postprocess_trajectory(traj, kw_args):
+
+            # Strip inactive DOFs from the trajectory.
+            openravepy.planningutils.ConvertTrajectorySpecification(
+                traj, config_spec
+            )
+
+            # Optionally execute the trajectory.
+            if 'execute' not in kw_args or kw_args['execute']:
+                return self.ExecuteTrajectory(traj, **kw_args)
+            else:
+                return traj
```

Comment:
We currently have three cases:

1. **Block, No Timeout:** wait forever (currently `timeout=None`, default)
2. **Block, Timeout:** wait for `timeout` seconds, then raise a `TimeoutException` (currently `timeout=5.0`)
3. **No Block:** return immediately (currently `timeout=0`)

I am very hesitant to switch the meaning of `None` from (1) to (3).

I'm also not sure why `defer=True` is a case. If we are going to make an API-breaking change (e.g. change the meaning of `timeout`), then we might as well choose the best API that we know of. When would you **not** want `ExecuteTrajectory` to return a future?